### PR TITLE
add testcase for #12005 

### LINKED ISSUE #12005: [ORC] `threadpool`'s `spawn(f(..))` won't compile when `f` has a return value...
With `newruntime`, when `threadpool`'s `spawn` returns a `FlowVar[T]` due to the `spawn`'ed call `proc` having a return value `T`, the call to `spawn` has a compiler error.

### Example
```nim
# `spawn` returning FlowVar thinks it needs `system.newObj` when it shouldn't...
# likely do to that FlowVar's used to require a `deepCopy` and thus `newObj`;
# but don't anymore with "newruntime"...

# compile with "--newruntime" and "--threads:on"

from threadpool import spawn, `^`, sync

proc doworkok(i: int) {.thread.} = echo i
spawn(doworkok(42)); sync() # this works when returning void!

proc doworkbad(i: int): int {.thread.} = i
echo ^spawn(doworkbad(42)) # this doesn't when returning a FlowVar[T]!
```

### Current Output
```
Error: system module needs: newObj
```

### Expected Output
It should compile and run without error.

### Additional Information
```
$ nim -v
Nim Compiler Version 0.20.99 HEAD/DEVEL nightly build of 19-August-2019 on Windows 64-bit
```

### PR DESCRIPTION
**Pull request by ringabout**

### CODE REVIEW

File: tests/stdlib/tthreadpool.nim [hunk 1]
```diff
@@ -0,0 +1,12 @@
+discard """
+  cmd: "nim $target --threads:on --gc:arc $options $file"
```
Comment (ringabout) [inline tests/stdlib/tthreadpool.nim]:
```suggestion
  cmd: "nim $target --threads:on --gc:orc $options $file"
```

File: tests/stdlib/tthreadpool.nim [hunk 2]
```diff
@@ -0,0 +1,12 @@
+discard """
+  cmd: "nim $target --threads:on --gc:orc $options $file"
```
Comment (timotheecour) [inline tests/stdlib/tthreadpool.nim]:
```suggestion
  matrix: "--threads:on --gc:orc"
```

simpler and equivalent?

File: tests/stdlib/tthreadpool.nim [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  output: "42"
+"""
+
+from std/threadpool import spawn, `^`, sync
+
+proc doworkok(i: int) {.thread.} = echo i
+spawn(doworkok(42)); sync() # this works when returning void!
+
+proc doworkbad(i: int): int {.thread.} = i
+doAssert ^spawn(doworkbad(42)) == 42
```
Comment (timotheecour) [inline tests/stdlib/tthreadpool.nim]:
```suggestion
block: # bug #12005
  proc doworkok(i: int) {.thread.} = echo i
  spawn(doworkok(42))
  sync() # this works when returning void!

  proc doworkbad(i: int): int {.thread.} = i
  doAssert ^spawn(doworkbad(42)) == 42 # bug was here
```

File: tests/stdlib/tthreadpool.nim (hunk 2 above)
Comment (ringabout) [inline tests/stdlib/tthreadpool.nim]:
```suggestion
  matrix: "--threads:on --gc:arc"
  disabled: "freebsd"
```

**Review by timotheecour — APPROVED** [review]

**Pull request merged**